### ISSUE TITLE
Extra file types are not displayed with SCIFIO on

### ISSUE THREAD

**Issue by karlduderstadt** [imagej/imagej#263]
IOPlugins provide a nice way of adding additional file types to imagej besides images. With pom 29.0.0 now the IOService contains a fallback to HandleExtraFileTypes (https://github.com/fiji/IO/commit/e41e251c1bb6451267dd01321728f800074322b2) so that using Open... in the File menu, or drag and drop over the imagej toolbar triggers opening of these extra file types.

This works great if SCIFIO is turned off. However, if SCIFIO is turned on the IOPlugin.open method is called and the files are opened, but they are never displayed. Basically, I think uiService.show(..) is never called. 

I think this call should be in the DefaultLegacyOpener found in net.imagej.legacy.plugin. Currently, only Datasets are shown. Would it be possible to add an else statement that calls uiService.show(...) for all types that are not Datasets in DefaultLegacyOpener? 

Maybe there is another place where this should happen and I missed it. One way to test would be to try to open a csv and see if a TableDisplay opens or not. Currently, nothing is opened if SCIFIO is turned on.

**Comment by hinerm** [comment]
@karlduderstadt @ctrueden it looks like the problem is that the SciJava IO plugin was always completely claiming the IO process ([returning a non-null value even if no handler was found](https://github.com/imagej/imagej-legacy/blob/d2a00f7f8779a717eb632a7d59e1fe047a191a4a/src/main/java/net/imagej/legacy/plugin/DefaultLegacyOpener.java#L132)).

I have a [PR that revises this behavior](https://github.com/imagej/imagej-legacy/pull/252) to allow ImageJ 1.x openers to still work (like .csv) - but wanted to run this by @ctrueden, as there might be unintended consequences of this change?

**Comment by imagejan** [comment]
@hinerm wrote:

> ImageJ 1.x openers to still work (like .csv)

In an up-to-date Fiji, we should have an [`IOPlugin`](https://github.com/scijava/scijava-plugins-io-table/blob/55f5be4fcb3b3e05fc7ff17c17c8d7c3ab79316c/src/main/java/org/scijava/table/DefaultTableIOPlugin.java) that handles csv files, so no need to fall back to ImageJ1.

(Also, as far as I understood @ctrueden's comments in the forum topic, the legacy image opener now falls back to SCIFIO, so we tested all possibilities in ImageJ1 already, before ending up here, no?)

I understood that the problem is `TableIOPlugin` returning `GenericTable`, but no `Dataset`. And only `Dataset`s are displayed by default in the UI. Did I misunderstand this?

**Comment by karlduderstadt** [comment]
@imagejan Yes, exactly. The problem is that only Dataset type is displayed and all IOPlugins for other types and opened but never displayed. I thought the csv was a good example because this should be displayed by the IOPlugins for scijava tables.

I have also added two IOPlugins with other file extensions and if SCIFIO is turned on, they are opened, but they are never displayed. I have to explicitly call uiService.show( in the open method to have them displayed.

Somewhere SCIFIO IOPlugin types that are not Datasets need to be shown using uiService.show(..). I mentioned the location where I think it should be above, but maybe I missed something and there is another place.

**Comment by hinerm** [comment]
@karlduderstadt I'm sorry, I totally misunderstood the problem. From what I can tell, I agree with you about the location of where to slot in the `uiService.show` call, and [opened a new PR reflecting that change](https://github.com/imagej/imagej-legacy/pull/253).

@imagejan thank you for clarifying! I was testing with just IJ2 instead of Fiji and confused myself. I am still wondering about the behavior of the `DefaultLegacyOpener` never returning a `null` value, and thus never allowing another opener to have a chance... it's [straying from the javadoc](https://github.com/imagej/imagej-legacy/blob/d2a00f7f8779a717eb632a7d59e1fe047a191a4a/src/main/java/net/imagej/legacy/plugin/LegacyOpener.java#L54-L55), but perhaps intentionally? I [made a separate patch](https://github.com/imagej/imagej-legacy/commit/f06818bc680cdbe5fc57d6ca3634f930fc77b2ab) to update the return values and wonder what you and @ctrueden think?

**Comment by karlduderstadt** [comment]
@hinerm Awesome! Thanks for opening the PR!
